engine/ui/gGUIToolbar.cpp

```diff
@@ -28,19 +28,20 @@ int gGUIToolbar::getToolbarType() {
 }
 
 void gGUIToolbar::draw() {
+	// The foregroundcolor and backgroundcolor assignments have been made.
 //	gLogi("gGUIToolbar") << "draw";
 //	gLogi("gGUIToolbar") << "l:" << left << ", t:" << top << ", w:" << width << ", h:" << height;
 	gColor* oldcolor = renderer->getColor();
 	if(toolbartype == TOOLBAR_HORIZONTAL) {
-		renderer->setColor(toolbarforegroundcolor);
+		renderer->setColor(*toolbarforegroundcoloaradress);
```

Comment:
This is also unnecessary, you can make toolbarforegroundcolor gColor* and remove the dereference operation (*)